catroid/src/main/java/org/catrobat/catroid/content/Sprite.java

```diff
@@ -377,6 +377,7 @@ public Sprite cloneForCloneBrick() {
 		cloneLook(cloneSprite);
 		setUserAndVariableBrickReferences(cloneSprite, userBricks);
 
+		ProjectManager.getInstance().addSprite(cloneSprite);
 		ProjectManager.getInstance().setCurrentSprite(originalSprite);
```

Comment:
just a quick thought here. weve already got massive memory leaks with clones and broadcasts (just check out the android monitor when running that airfight game). so if theres sprites added to yet another list somewhere, are they ever going to be removed again?

Reply:
Thanks for the suggestion, i have been trying to track this List but it doesn't seem that it is ever cleaned. This means that not only the elements that i add but also the elements previously in the list are never removed, just dropped when a new project is initialized.

Reply:
I'm not sure what _this_ list is, but if you're talking about the sprite list of scenes/project
https://github.com/Catrobat/Catroid/blob/b7ad30d730e041a2e1828834cda4d9d09dc7142b/catroid/src/main/java/org/catrobat/catroid/stage/StageListener.java#L305-L309

Reply:
My bad, totally missed that, thanks for pointing it out. I tested it now and yes, after the execution of the program, when the dispose action is triggered the cloned Sprites that are now added in the Sprite class are removed from the List.

Reply:
No problem! :wink:

I'm not a fan of populating the **real** sprites list of every scene with just temporary objects. These objects in the list are the objects shown in the IDE and also are serialized. But hey, everything is already implemented that way…

Reply:
> But hey, everything is already implemented that way…

or not. Don't know :smile:  Just assuming because the method `disposeClonedSprites` exists.